frontend/src/modules/app/ToolBar.tsx

```diff
@@ -243,76 +229,73 @@ export const ToolBar = ({
       case VentilationMode.niv_pc:
       case VentilationMode.niv_ps:
       case VentilationMode.psv:
-        dispatch(
-          commitRequest<ParametersRequest>(MessageType.ParametersRequest, {
-            peep: parameterRequestDraft.peep,
-            vt: parameterRequestDraft.vt,
-            rr: parameterRequestDraft.rr,
-            fio2: parameterRequestDraft.fio2,
-          }),
-        );
+        dispatchParameterRequest({
+          peep: parameterRequestDraft.peep,
+          vt: parameterRequestDraft.vt,
+          rr: parameterRequestDraft.rr,
+          fio2: parameterRequestDraft.fio2,
+        });
         break;
       default:
         break;
     }
-  }, [isVentilatorOn, parameterRequestDraft, alarmLimitsRequestDraft, currentMode, dispatch]);
+  };
 
   /**
-   * Disabled Start/Pause Ventilation button when backend connection is lost
+   * Set Label for Start/Pause Ventilation button
    */
   useEffect(() => {
-    if (storeReady) {
-      setLandingLabel('Start');
-      setIsDisabled(false);
-      setLabel(ventilating ? 'Pause Ventilation' : 'Start Ventilation');
-    } else if (ventilatingStatus) {
-      setIsDisabled(true);
+    if (ventilatingStatus) {
       setLabel('Connecting...');
     } else {
-      setLandingLabel('Loading...');
-      setIsDisabled(true);
-      setLabel('Loading...');
+      setLabel(ventilating ? 'Pause Ventilation' : 'Start Ventilation');
     }
   }, [storeReady, ventilatingStatus, ventilating]);
 
-  useEffect(() => {
-    if (ventilating) {
-      return;
-    }
-
-    initParameterUpdate();
-  }, [ventilating, initParameterUpdate]);
-
   /**
-   * Update Label on Button based on ventilation status
+   * Switch to dashboard page if ventilating
    */
   useEffect(() => {
-    if (ventilating) {
+    if (storeReady && ventilating) {
       history.push(DASHBOARD_ROUTE.path);
     }
-    setIsVentilatorOn(ventilating);
-  }, [ventilating, history]);
+  }, [ventilating, history, storeReady]);
 
-  const StartPauseButton = (
+  const StartPauseVentilationButton = (
```

Comment:
The button and its behavior is also a good candidate for decomposing out to a separate component, since it has some nontrivial behavior of its own which probably doesn't/shouldn't depend on the rest of the toolbar. This would be for a future refactoring PR.